### PR TITLE
Internal Clock Enhancements

### PR DESCRIPTION
**Pull request by djphazer**
This is a fairly elegant solution to #23 that's hopefully not too intrusive.

Clock Forwarding and internal clock Start/Stop are now independent settings. On the ClockSetup screen, turn encoder left to toggle Start/Stop, turn right to toggle Forwarding, indicated by a LINK_ICON. Long-press left encoder will still Pause/Unpause.
Internal clock always triggers Digital 1, will only trigger Digital 3 if forwarding is ON.

Functionally, I think it's exactly what I wanted. Open to feedback on the interface screen.

### CODE REVIEW

**Comment by djphazer** [comment]
After sufficient time and testing, I've rebased and squashed this down into a single commit. It's quite functional, but some code review would be a good idea...

Also, it probably depends on #59 to fit in memory

File: software/o_c_REV/HEM_ClockSetup.ino [hunk 1]
```diff
@@ -91,44 +108,91 @@ protected:
     }
 
 private:
-    int cursor; // 0=Source, 1=Tempo, 2=Multiply
+    char cursor; // 0=Play/Stop, 1=Forwarding, 2=External PPQN, 3=Tempo, 4,5=Multiply
```
Comment (Spyes) [inline software/o_c_REV/HEM_ClockSetup.ino]: Might be worthwhile to extract this to an enum, it'll make it much easier to understand at a glance the different cursor states

File: software/o_c_REV/HSClockManager.h [hunk 2]
```diff
@@ -24,34 +24,36 @@
...
+    uint32_t clock_tick = 0; // tick when a physical clock was received on DIGITAL 1
+    uint32_t beat_tick[3] = {0,0,0}; // The tick to count from
+    uint32_t last_tock_check[3] = {0,0,0}; // To avoid checking the tock more than once per tick
+    bool tock[3] = {0,0,0}; // The current tock value
+    int tocks_per_beat[3] = {1, 1, 24}; // Multiplier
+    int clock_ppqn = 4; // external clock multiple
+    bool cycle = 0; // Alternates for each tock, for display purposes
+    int count[3] = {0,0,0}; // Multiple counter, 0 is a special case when first starting the clock
+
+    bool boop[4]; // Manual triggers
```
Comment (Spyes) [inline software/o_c_REV/HSClockManager.h]: consider extracting hard-coded array lengths to consts

**Review by Spyes — COMMENTED** [review]
Absolutely solid feature!!

File: software/o_c_REV/HEM_ClockSetup.ino (hunk 1 above)
Comment (djphazer) [reply software/o_c_REV/HEM_ClockSetup.ino]: Yeah, and there's probably a cursor variable in every other applet that should be an enum... I'll work on this.

**Pull request closed without merging**